FOI: 3.9 with h5py 2.10.0 started to fail to import

`ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject`

```
Run python -m pytest -s -v --cov=dandi --cov-report=xml dandi
ImportError while loading conftest '/home/runner/work/dandi-cli/dandi-cli/dandi/conftest.py'.
dandi/conftest.py:1: in <module>
    from .tests.fixtures import *  # noqa: F401, F403
dandi/tests/fixtures.py:12: in <module>
    import pynwb
/opt/hostedtoolcache/Python/3.9.1/x64/lib/python3.9/site-packages/pynwb/__init__.py:7: in <module>
    import h5py
/opt/hostedtoolcache/Python/3.9.1/x64/lib/python3.9/site-packages/h5py/__init__.py:46: in <module>
    from ._conv import register_converters as _register_converters
h5py/_conv.pyx:1: in init h5py._conv
    ???
h5py/h5t.pyx:1: in init h5py.h5t
    ???
E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
Error: Process completed with exit code 4.
```
my wild guess is that due to pynwb limiting h5py to 2.x series, but we did test/impor ok just recently with such restrictions.  Did not look in detail on what is going on and where to complain - I do not think it is something we could address on our end.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -55,6 +55,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip wheel
+        pip install 'numpy<1.19.4'
         pip install ".[test]"
 
     - name: Set environment variable when using Python 3.6 to cover more code

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -55,6 +55,13 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip wheel
+        # As of 2021-02-01, the latest version of hdmf (v2.3.0) requires numpy
+        # >=1.16, <1.19.4, so we end up with numpy 1.19.3 installed.  However,
+        # due to the release of numpy 1.20.0 on January 30, h5py gets built
+        # against the later version instead, and because numpy 1.20.0 changed
+        # the size of ndarray, an error results at runtime.  This can be worked
+        # around by installing numpy before the other dependencies so that it
+        # is available when h5py is built.
         pip install 'numpy<1.19.4'
         pip install ".[test]"
 

